### DISCUSSION TITLE
Does openssl has support for FIPS 186-5 rsa key generation?

### DISCUSSION THREAD

**Discussion by rahulcode60** [openssl/openssl#25215] [Q&A]
Does openssl has support for FIPS 186-5 rsa key generation?

If I use 186-5 revision algorithm vectors to test on openssl apis I get this error.
**4087FAF7FF7F0000:error:020000B2:rsa routines:ossl_rsa_fips186_4_gen_prob_primes:pub exponent out of range:crypto/rsa/rsa_sp800_56b_gen.c:97:**

I can see in fips 186-5 they stated.
**Range for the Public Exponent (e): The public exponent e must be an odd, positive integer, and its value must lie within the range 
2 ^16<e<2 ^256
 
 . This constraint is imposed to ensure the security of the RSA algorithm while maintaining efficient computation during encryption and signature verification.**

**Comment by nhorman** [comment]
Strictly speaking, openssl support FIPS186-4 (as the function name in the error above suggests).  However, the check above is valid in both 186-4 and 186-5, in that e must be between 2^16<e<2^256

What is the value of e in the test vector you are using above?  Is this perhaps a negative test in your suite (i.e. confirming failure for an out of bounds exponent)?

**Reply by rahulcode60** [reply]
I am using this value.

"0779DCE7AB"

For public key exponent.

"50884093323" in decimal
